zookeeper-server/src/test/java/org/apache/zookeeper/server/RequestThrottlerTest.java

```
@@ -311,34 +311,38 @@ public void testLargeRequestThrottling() throws Exception {
 
-        submitted.await(5, TimeUnit.SECONDS);
-
-        resumeProcess.countDown();
-
-        // We should start throttling instead of queuing more requests.
-        //
-        // We always allow up to GLOBAL_OUTSTANDING_LIMIT + 1 number of requests coming in request processing pipeline
-        // before throttling. For the next request, we will throttle by disabling receiving future requests but we still
-        // allow this single request coming in. So the total number of queued requests in processing pipeline would
-        // be GLOBAL_OUTSTANDING_LIMIT + 2.
-        assertEquals(Integer.parseInt(GLOBAL_OUTSTANDING_LIMIT) + 2,
-                (long) MetricsUtils.currentServerMetrics().get("prep_processor_request_queued"));
+            for (int i = 0; i < totalRequests; i++) {
+                zk.create("/request_throttle_test- " + i, ("/request_throttle_test- "
+                        + i).getBytes(), ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT, (rc, path, ctx, name) -> {
+                }, null);
+            }
 
-        System.clearProperty(ZooKeeperServer.GLOBAL_OUTSTANDING_LIMIT);
+            submitted.await(5, TimeUnit.SECONDS);
+
+            resumeProcess.countDown();
+
+            // We should start throttling instead of queuing more requests.
+            //
+            // We always allow up to GLOBAL_OUTSTANDING_LIMIT + 1 number of requests coming in request processing pipeline
+            // before throttling. For the next request, we will throttle by disabling receiving future requests but we still
+            // allow this single request coming in. So the total number of queued requests in processing pipeline would
+            // be GLOBAL_OUTSTANDING_LIMIT + 2.
+            assertEquals(Integer.parseInt(GLOBAL_OUTSTANDING_LIMIT) + 2,
+                    (long) MetricsUtils.currentServerMetrics().get("prep_processor_request_queued"));
+        } catch (Exception e) {
```

ops, this is not good.
I missed this part.
if you catch every Exception you are making JUnit ignore errors

---

I can amend your commit @hanm if you do not have time.
I would like to prepare the RC0 of 3.6.1 today or tomorrow

sorry I missed this problem

---

fixed by rethrowing.